modules/light-clients/07-tendermint/types/update.go

```diff
@@ -162,85 +162,124 @@ func checkTrustedHeader(header *Header, consState *ConsensusState) error {
 	return nil
 }
 
-// checkValidity checks if the Tendermint header is valid.
-// CONTRACT: consState.Height == header.TrustedHeight
-func checkValidity(
-	clientState *ClientState, consState *ConsensusState,
-	header *Header, currentTimestamp time.Time,
+// ValidateClientMessage checks if the clientMessage is of type Header or Misbehaviour and validates the message
+func (cs *ClientState) ValidateClientMessage(
```

Comment:
@colin-axner @damiannolan Open to ideas regarding the fn parameters here. It was necessary to pass in `ctx`, `cdc`, `clientStore` for the `Misbehaviour` type but perhaps there is a way around this.

Reply:
It would be nice to settle on the APIs of the new client state methods. It's likely that 06-solomachine will not make use of all the args that 07-tendermint does, but it will still require them to be provided to fulfil the expected interface.

Reply:
It should be the same as the existing interface [function](https://github.com/cosmos/ibc-go/blob/e0ae0cd84433113a1016f12f2d478e128b244fa2/modules/core/exported/client.go#L61)